controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/integration/noderepository/NodeRepositoryNode.java

```diff
@@ -433,6 +424,19 @@ public void setSwitchHostname(String switchHostname) {
         this.switchHostname = switchHostname;
     }
 
+
+    // --- Helper methods for code that (wrongly) consume this directly
+
+    public boolean hasType(NodeType type) {
+        return type.name().equals(getType());
+    }
+
+    public boolean hasState(NodeState state) {
+        return state.name().equals(getState());
+    }
+
+    // --- end
+
```

Comment:
Are these used anywhere?

Reply:
Yes, some internal code still uses this directly.